### PR TITLE
Sample showing integration using the SQL transport

### PR DESCRIPTION
**Pull request by andreasohlund**
@Particular/documentation please review

### CODE REVIEW

**Comment by mauroservienti** [comment]
First question: why **I** added the review label? :-)

image

**Comment by mauroservienti** [comment]
Shouldn't we exclude ".DotSettings" files from commits?

File: samples/sqltransport/native-integration/sample.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+---
+title: SQL Server
+summary: 'How to integrate SQL Server transport with NHibernate persistence without outbox'
+tags:
+- SQL Server
+- NHibernate
+---
+
+ 1. Make sure you have SQL Server Express installed and accessible as `.\SQLEXPRESS`. Create a databases `samples`
```
Comment (mauroservienti) [inline samples/sqltransport/native-integration/sample.md]: misses the last full stop after "samples"

File: samples/sqltransport/native-integration/sample.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+title: SQL Server
+summary: 'How to integrate SQL Server transport with NHibernate persistence without outbox'
+tags:
+- SQL Server
+- NHibernate
+---
+
+ 1. Make sure you have SQL Server Express installed and accessible as `.\SQLEXPRESS`. Create a databases `samples`
+ 2. Start the Receiver project.
+ 3. In the Sender's console you should see `Press any key to send a message. Press `q` to quit` text when the app is ready.
```
Comment (mauroservienti) [inline samples/sqltransport/native-integration/sample.md]: Sender and not Sender's, as far as I can tell

File: samples/sqltransport/native-integration/sample.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+tags:
+- SQL Server
+- NHibernate
+---
+
+ 1. Make sure you have SQL Server Express installed and accessible as `.\SQLEXPRESS`. Create a databases `samples`
+ 2. Start the Receiver project.
+ 3. In the Sender's console you should see `Press any key to send a message. Press `q` to quit` text when the app is ready. 
+ 4. Hit any key.
+ 5. A message will be sent using ADO.net and be received by the app.
```
Comment (mauroservienti) [inline samples/sqltransport/native-integration/sample.md]: I'd separate the following steps from the one from 1 to 5, they are a different sample to run

File: samples/sqltransport/native-integration/sample.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+
+<!-- import EndpointConfiguration-->
+
+
+The table would be `Samples.SqlServer.NativeIntegration` in the database `samples` on server `.\SQLEXPRESS` (localhost). Now that we know where to put the data lets see how we can make sure that the endpoint can parse our message.
+
+### Serialization
+
+In this sample we'll be using JSON to serialize the messages but XML would have worked equally well. First we'll setup the endpoint to user JSON using the following configuration:
```
Comment (mauroservienti) [inline samples/sqltransport/native-integration/sample.md]: "to use" not "to user"

File: samples/sqltransport/native-integration/sample.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+<!-- import SerializerConfiguration-->
+
+Not that our endpoint can understand JSON we define a message contract to use. In NServiceBus messages are just C# classes so in this case we create the following class
```
Comment (mauroservienti) [inline samples/sqltransport/native-integration/sample.md]: colon at the end

File: samples/sqltransport/native-integration/sample.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+<!-- import MessageContract-->
+
+The final piece of the puzzle to to tell the serializer what C# class our JSON payload belongs to. We do this using the JSON.net `$type` attribute. The message body will then look as follows:
```
Comment (mauroservienti) [inline samples/sqltransport/native-integration/sample.md]: The final piece of the puzzle **is** to tell the serializer

File: samples/sqltransport/native-integration/sample.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+<!-- import MessagePayload-->
+
+
+### Sending the message
+Not that we've done all the legwork send a message to our endpoint using plain ADO.net is as easy as
```
Comment (mauroservienti) [inline samples/sqltransport/native-integration/sample.md]: I don't understant

File: samples/sqltransport/native-integration/sample.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+<!-- import SendingUsingAdoNet-->
+
+Armed with this you will now be able to send messages from any app in your organization that supports ADO.net
```
Comment (mauroservienti) [inline samples/sqltransport/native-integration/sample.md]: Armed with this now you will be able to send messages, in your organization, from any app that supports ADO.net.

**Comment by andreasohlund** [comment]
The ´Review´ label comes from the waffle.io auto move pull to X feature

**Comment by andreasohlund** [comment]
.DotSettings is needed to make sure we don't use `var` since var make the free text search not work that well

**Comment by SimonCropp** [comment]
@andreasohlund but why does waffle pick random people to perform actions under?

File: samples/sqltransport/native-integration/Version_2/Receiver/Program.cs [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    static void Main()
+    {
+        #region EndpointConfiguration
+        BusConfiguration busConfig = new BusConfiguration();
+
+        busConfig.UseTransport<SqlServerTransport>()
+            .ConnectionString(@"Data Source=.\SQLEXPRESS;Initial Catalog=samples;Integrated Security=True");
+        busConfig.EndpointName("Samples.SqlServer.NativeIntegration");
+        #endregion
+        #region SerializerConfiguration
```
Comment (SimonCropp) [inline samples/sqltransport/native-integration/Version_2/Receiver/Program.cs]: i think i would just dump SerializerConfiguration inside EndpointConfiguration

File: samples/sqltransport/native-integration/sample.md [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+and create an `on inserted` trigger that will send a `LegacyOrderDetected` message for each new order that we add to the table. Here's the trigger:
+
+<!-- import CreateTrigger -->
+
+Notice how we generate a unique message id by hashing the identity column. NServiceBus requires each message to have a unique id in order to safely perform retries.
+
+That's it, just add a few orders to the table and notice how the app receives the messages. In a real life scenario you would likely use this to trigger a `bus.Publish` to push an `OrderAccepted` event out on the bus.
+
+Happy inserting!
+
```
Comment (SimonCropp) [inline samples/sqltransport/native-integration/sample.md]: lots of whitespace :)

File: samples/sqltransport/native-integration/sample.md [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+and create an `on inserted` trigger that will send a `LegacyOrderDetected` message for each new order that we add to the table. Here's the trigger:
+
+<!-- import CreateTrigger -->
+
+Notice how we generate a unique message id by hashing the identity column. NServiceBus requires each message to have a unique id in order to safely perform retries.
+
+That's it, just add a few orders to the table and notice how the app receives the messages. In a real life scenario you would likely use this to trigger a `bus.Publish` to push an `OrderAccepted` event out on the bus.
+
+Happy inserting!
```
Comment (SimonCropp) [inline samples/sqltransport/native-integration/sample.md]: i don think the tone of `Happy inserting!` in consistent with our other samples

**Comment by mauroservienti** [comment]
I suppose it is not random at all, I suppose it is the person that configured that action / column behavior in waffle the first time. I did the review by PR one.

-----Original Message-----

**Comment by danielmarbach** [comment]
Nicely done. I think this will help a lot of customers

**Comment by andreasohlund** [comment]
Updated after review feedback

**Pull request merged**